src/bin/tikv-server.rs

```diff
@@ -62,139 +63,107 @@ fn print_usage(program: &str, opts: Options) {
     print!("{}", opts.usage(&brief));
 }
 
-fn get_string_value<F>(short: &str,
-                       long: &str,
-                       matches: &Matches,
-                       config: &toml::Value,
-                       default: Option<String>,
-                       f: F)
-                       -> String
-    where F: Fn(&toml::Value) -> Option<String>
-{
-    // avoid panic if short is not defined.
-    let s = if matches.opt_defined(short) {
-        matches.opt_str(short)
-    } else {
-        None
-    };
+fn get_flag_string(matches: &Matches, name: &str) -> Option<String> {
+    if matches.opt_defined(name) {
+        let s = matches.opt_str(name);
+        info!("flag {}: {:?}", name, s);
 
-    s.or_else(|| {
-            config.lookup(long).and_then(|v| f(v)).or_else(|| {
-                info!("{}, use default {:?}", long, default);
-                default
-            })
-        })
-        .expect(&format!("please specify {}", long))
+        return s;
+    }
+    None
 }
 
-fn get_boolean_value<F>(short: &str,
-                        long: &str,
-                        matches: &Matches,
-                        config: &toml::Value,
-                        default: Option<bool>,
-                        f: F)
-                        -> bool
-    where F: Fn(&toml::Value) -> Option<bool>
-{
-    let b = if matches.opt_defined(short) {
-        matches.opt_str(short).map(|x| x.parse::<bool>().unwrap())
-    } else {
-        None
-    };
+fn get_flag_int(matches: &Matches, name: &str) -> Option<i64> {
+    if matches.opt_defined(name) {
+        let i = matches.opt_str(name).map(|x| {
+            x.parse::<i64>()
+                .or_else(|_| util::config::parse_readable_int(&x))
+                .unwrap_or_else(|e| {
+                    warn!("parse {} failed: {:?}", name, e);
+                    process::exit(1)
+                })
+        });
+        info!("flag {}: {:?}", name, i);
 
-    b.or_else(|| {
-            config.lookup(long).and_then(|v| f(v)).or_else(|| {
-                info!("{}, use default {:?}", long, default);
-                default
-            })
+        return i;
+    }
+    None
+}
+
+fn get_toml_boolean(config: &toml::Value, name: &str, default: Option<bool>) -> bool {
+    let b = config.lookup(name)
+        .and_then(|v| v.as_bool())
```

Comment:
Then `v.as_bool() ` returns `None`, it tries to unwrap `default`. If `default` is `None` too, it exits with 1.